libfreerdp/core/transport.c

```diff
@@ -779,7 +800,27 @@ int transport_read_pdu(rdpTransport* transport, wStream* s)
 	if (!Stream_EnsureCapacity(s, Stream_GetPosition(s) + pduLength))
 		return -1;
 
-	status = transport_read_layer_bytes(transport, s, pduLength - Stream_GetPosition(s));
+	if (!UseExternalIO)
```

Comment:
why not simply hook in `transport_read_layer_bytes`?

Reply:
`transport_read_pdu` fit better for task, as we need to check for complete pdu, but do not need to read by self, so we need to run all checks from `transport_read_pdu` against buffer filled by external io